python/main.py

```diff
@@ -255,4 +256,89 @@ def update_status(id):
 
     return item_details(id)
 
+@app.post("/auction/{item_id}", status_code=201)
+def add_bid(item_id: str, bid_price: str = Form(...)):
+    logger.info(f"New bid: {bid_price} yen for items_id: {item_id}")
+
+    conn = sqlite3.connect(dbname)
+    c = conn.cursor()
+    
+    try:
+        # generate UUID
+        generated_id = str(uuid.uuid4())
+    
+        c.execute("SELECT name FROM items WHERE id = (?)", (item_id,))
+        data = c.fetchall()
+        
+        (name, ) = data[0]
+        print(name)
+        c.execute(
+            '''
+            INSERT INTO
+                auction (id, bidder_name, items_id, bid_price, item_name)
+            VALUES 
+                (?, ?, ?, ?, ?)
+            ''',
+            (generated_id, "Bidder 1", item_id, bid_price, name)
+        )
+        conn.commit()
+    except sqlite3.IntegrityError as err:
+        if "UNIQUE constraint" in str(err):
+            raise HTTPException(status_code=400, detail=f"This user already has bid for this item")
+            logger.error(f"Unhandled error occurred")
+            raise err
+    finally:
+        conn.close()
+
+    return {data[0]}
+    #return {"message": f"item received: {name}"}
+
+@app.get("/auction/{item_id}")
+def show_auction(item_id):
+    conn = sqlite3.connect(dbname)
+    conn.row_factory = sqlite3.Row
+    c = conn.cursor()
+
+    c.execute(
+        '''
+        SELECT 
+            bidder_name,
+            bid_price
+        FROM 
+            auction 
+        WHERE
+            items_id = ?
+        ''',
+        (item_id,)
+    )
+    response = { "items": [row for row in c] }
+    conn.close()
+
+    return response
+
+@app.put("/auction/{item_id}")
+def update_bid(item_id: str, bid_price: str = Form(...)):
```

Comment:
@dominic-soh 
Should update_bid check if `bid_price > price` or is that frontend's job?

Reply:
I think i will make frontend validate this